Documentation/operation docs rework (#3074)
scality/metalk8s: docs/operation/solutions.rst

Solution Deployment
===================

To deploy a Solution in a MetalK8s cluster, a utility script is provided.

> This procedure describes how to deploy a Solution using this

Reviewer: ```suggestion
This procedure describes how to deploy a solution using this
```

> tool located at the root of MetalK8s archive:

Reviewer: ```suggestion
tool, which is located at the root of the MetalK8s archive:
```

  .. parsed-literal::

    /srv/scality/metalk8s-|version|/solutions.sh

Preparation
-----------

> #. Import a Solution in the cluster, and make the container images

Reviewer: ```suggestion
#. Import a solution in the cluster, and make the container images
```

   available through the cluster registry.

   .. code::

      ./solutions.sh import --archive </path/to/solution.iso>

> #. Activate a Solution version.

Reviewer: ```suggestion
#. Activate a solution version.
```

   .. code::

      ./solutions.sh activate --name <solution-name> --version <solution-version>

>    Only one version of a Solution can be active at any point in time.

Reviewer: ```suggestion
   Only one version of a solution can be active at a time.
```

>    An active Solution version provides the cluster-wide resources,

Reviewer: ```suggestion
   An active solution version provides cluster-wide resources,
```

>    such as CRDs, to all other versions of this Solution.

Reviewer: ```suggestion
   such as CRDs, to all other versions of this solution.
```

Deployment
----------

#. Solutions are meant to be deployed in isolated namespaces called

>    :term:`Environments <Environment>`.

Reviewer: ```suggestion
   :term:`environments <Environment>`.
```

>    To create an Environment, run:

Reviewer: ```suggestion
   To create an environment, run:
```

   .. code::

      ./solutions.sh create-env --name <environment-name>

#. Solutions are packaged with an :term:`Operator`, and optionally an
   associated web UI, to provide all required domain-specific logic.

>    To deploy a Solution Operator and its UI in an Environment, run:

Reviewer 2: @gdemonet @alexandre-allard-scality to confirm but I think we don't deploy the admin UI anymore when doing add solution

Reviewer 3: Indeed, could be:

```suggestion
   To deploy a Solution Operator in an Environment, run:
```

>    To deploy a Solution Operator and its UI in an Environment, run:

Reviewer: ```suggestion
   To deploy a solution operator and its UI in an environment, run:
```

   .. code::

      ./solutions.sh add-solution --name <environment-name> \
      --solution <solution-name> --version <solution-version>

Configuration
-------------

The Solution Operator and UI (if any) are now deployed.

> To finalize the deployment and configuration of the Solution application,

Reviewer: ```suggestion
To finalize the deployment and configuration of a solution,
```

> refer to its documentation specifics.

Reviewer: ```suggestion
refer to its documentation.
```

Solution Deployment
===================

To deploy a solution in a MetalK8s cluster, a utility script is provided.
This procedure describes how to deploy a solution using this
tool, which is located at the root of the MetalK8s archive:

  .. parsed-literal::

    /srv/scality/metalk8s-|version|/solutions.sh

Preparation
-----------

#. Import a solution in the cluster, and make the container images
   available through the cluster registry.

   .. code::

      ./solutions.sh import --archive </path/to/solution.iso>

#. Activate a solution version.

   .. code::

      ./solutions.sh activate --name <solution-name> --version <solution-version>

   Only one version of a solution can be active at a time.
   An active solution version provides cluster-wide resources,
   such as CRDs, to all other versions of this solution.

Deployment
----------

#. Solutions are meant to be deployed in isolated namespaces called
   :term:`environments <Environment>`.

   To create an environment, run:

   .. code::

      ./solutions.sh create-env --name <environment-name>

#. Solutions are packaged with an :term:`Operator`, and optionally an

>    associated web UI, to provide all required domain-specific logic.

Reviewer 3: ```suggestion
#. Solutions are packaged with an :term:`Operator` to provide all
   required domain-specific logic.
```

   To deploy a solution operator in an environment, run:

   .. code::

      ./solutions.sh add-solution --name <environment-name> \
      --solution <solution-name> --version <solution-version>

Configuration
-------------

> The solution operator and UI (if any) are now deployed.

Reviewer 3: ```suggestion
The solution operator is now deployed.
```
